#797 followup: `ArchHooks_Win32Static` housekeeping

sukibaby opened the pull request:
re: https://github.com/itgmania/itgmania/pull/797#issuecomment-2848655576

### List of changes:

- File reorginization 
- `LangIdToString(LANGID lang_id)` no longer  requires `GetLanguageID()`, since we don't need to load `kernel32.dll` to call `GetUserDefaultLangID()` since Windows Vista, so the entire function `GetLanguageID()` is removed
- Added missing `LL` specifier to QueryPerformanceCounter math to avoid an unwanted type conversion
- Remove extraneous comments

sukibaby commented:
sorry for all the extra pushes here. My IDE is acting bizarre on style formatting suddenly

teejusb commented on src/arch/ArchHooks/ArchHooks_Win32Static.cpp:
```diff
@@ -10,59 +10,160 @@
+	// in the QuadPart of the LARGE_INTEGER, which is a 64-bit integer.
+	static LARGE_INTEGER g_frequency;
+
+	static void InitTimer()
```
The functions and variables don't need to be marked as static as they already have static linkage from being in an anonymous namespace

sukibaby replied:
appreciate it, wasn't confident on the stylistic convention there

teejusb commented on src/arch/ArchHooks/ArchHooks_Win32Static.cpp:
```diff
@@ -10,59 +10,160 @@
-    LARGE_INTEGER g_liFrequency;
-    LARGE_INTEGER g_liCurrentTime;
-}  // namespace
+	static std::once_flag g_timer_init_flag;
```
Variables with static linkage actually follow the kSnakeCase format instead of snake_case:

https://google.github.io/styleguide/cppguide.html#Constant_Names

I'm okay keeping it at g_timerInitFlag or we rename it to the follow the google convention.

teejusb commented on src/arch/ArchHooks/ArchHooks_Win32Static.cpp:
```diff
@@ -10,59 +10,160 @@
 
 	// Get the current time.
-	QueryPerformanceCounter(&g_liCurrentTime);
+	LARGE_INTEGER current_time;
```
I think the benefit of this being statically scoped is that you don't have to do an allocation/release every time `GetSystemTimeInMicroseconds` is called and can use the same variable which I think would save some cycles.

sukibaby replied:
True, that was my reasoning for keeping the `current_time` variable static when I proposed the implementation of QueryPerformanceCounter which we currently have.

When I profile in-place filling a static variable against filling a temporary variable, creating a temporary variable has a small advantage. It's a minor difference, but seems to indicate the overhead of accessing a static variable's place in memory outweighs the overhead of incurring an allocation/release.

sukibaby commented:
Noticed commit history was a bit of a mess here. Cleaned it up so it's nice and descriptive.

sukibaby committed:
Reorganize ArchHooks_Win32Static

Move all the static functions and variables inside an anonymous namespace.

Remove GetLanguageID because we don't need to load kernel32.dll to call GetUserDefaultLangID (this isn't Windows XP)

Remove a lot of unneeded comments while stuff is being moved.

sukibaby commented:
The following changes were just removed from this PR:
 - Renaming of variables
 - The `current_time` variable has been left where it was, for now

Now it is just the reorginization of static functions into the anonymous namespace and removal of an unneeded `kernel32.dll` loading.
